crates/bevy_render/src/mesh/shape/shapegen/interpolation.rs

```diff
@@ -0,0 +1,778 @@
+        &mut self.lerp
+    }
+    fn int2(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn int3(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn int4(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint2(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint3(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint4(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn float(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn float2(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn float3(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn float4(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uchar4norm(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+}
+
+///
+/// Two, three, and four component floating point
+/// attributes are spherically interpolated around
+/// `(0, 0, 0)`. Everything else is linearly interpolated.
+///
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct SphereInterpolatorGroup {
+    slerp: SphericalInterpolator,
+    lerp: LinearInterpolator,
+}
+
+impl Interpolator for SphereInterpolatorGroup {
+    type Int = LinearInterpolator;
+    type Int2 = LinearInterpolator;
+    type Int3 = LinearInterpolator;
+    type Int4 = LinearInterpolator;
+    type Uint = LinearInterpolator;
+    type Uint2 = LinearInterpolator;
+    type Uint3 = LinearInterpolator;
+    type Uint4 = LinearInterpolator;
+    type Float = LinearInterpolator;
+    type Float2 = SphericalInterpolator;
+    type Float3 = SphericalInterpolator;
+    type Float4 = SphericalInterpolator;
+    type Uchar4Norm = LinearInterpolator;
+
+    fn int(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn int2(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn int3(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn int4(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint2(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint3(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn uint4(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn float(&mut self, _: &str) -> &mut LinearInterpolator {
+        &mut self.lerp
+    }
+    fn float2(&mut self, _: &str) -> &mut SphericalInterpolator {
```

Comment:
Note that this could be incorrect: it would spherically interpolate 2D attributes which doesn't make sense for something like UV coordinates. I believe that changing this to be a linear interpolator would make more sense.